### ISSUE TITLE
Cascadia Mono font not found

### ISSUE THREAD

**Issue by HunterZ** [microsoft/terminal#16058]
### Windows Terminal version

1.18.2681.0

### Windows build number

10.0.22621.2283

### Other Software

winget v1.6.2701

### Steps to reproduce

Install an older version of Terminal. Upgrade to 1.18.2681.0 using winget.

### Expected Behavior

Upgrade succeeds.

### Actual Behavior

Terminal says it can't find a font named Cascadia Mono, and falls back to Consolas

**Comment by HunterZ** [comment]
I know this has been reported a million times, but it just happened to me and all of the previous issues on it seem to be closed, so it clearly isn't resolved.

I do not see Cascadia Mono in the Windows Fonts UI or in C:\Windows\Fonts or in %localappdata%\Microsoft\Windows\Fonts, so something either uninstalled it, or it is not included in Windows 11 Home.

I tried doing a Repair, but it didn't accomplish anything.

**Comment by zadjii-msft** [comment]
This is wacky and very unexpected. The other issues are all closed, because we _thought_ we fixed this a few releases ago. We added a final fallback to try and load the font directly from a file in our package (not from the font cache).

What rendering engine are you using (could you share your [settings.json file](https://github.com/microsoft/terminal/wiki/Frequently-Asked-Questions-(FAQ)#where-can-i-find-the-settings-file))? (we're pretty sure the rendering engine _shouldn't_ matter here.) 


Our only theory right now is that the way we detect a failure to load the font might not be working anymore?

23d45a7e3a286f37924fd68d222168fce631282d was the last touch here, which only shipped in 1.18 preview. That we _might_ need to revert 😞

**Comment by duckinator** [comment]
I just encountered this after upgrading using winget.

* Windows Terminal version (after upgrading): 1.18.2822.0
* Windows build number: 10.0.19045.0
* winget version: v1.6.2771

**Comment by vpal12** [comment]
yes

**Comment by HunterZ** [comment]
> This is wacky and very unexpected. The other issues are all closed, because we _thought_ we fixed this a few releases ago. We added a final fallback to try and load the font directly from a file in our package (not from the font cache).
> 
> What rendering engine are you using (could you share your [settings.json file](https://github.com/microsoft/terminal/wiki/Frequently-Asked-Questions-(FAQ)#where-can-i-find-the-settings-file))? (we're pretty sure the rendering engine _shouldn't_ matter here.)
[settings.json](https://github.com/microsoft/terminal/files/12907483/settings.json)

Reminder that Cascadia Mono did not seem to be installed in Windows 11 for some reason (which is weird because I was using an old version of Terminal without having ever seen that error), so it seems to be Terminal depending on something which isn't guaranteed to be present?

**Comment by duckinator** [comment]
Here's my [settings.json](https://github.com/microsoft/terminal/files/12908917/settings.json), too.

I've not reconfigured anything related to rendering engines, fonts, or anything like that.

**Comment by duckinator** [comment]
The plot thickens: I have `C:\Windows\Fonts\CascadiaMono.ttf`, and I was also able to use the Cascadia Mono font in Notepad without issue.

![image](https://github.com/microsoft/terminal/assets/39698/073f6820-92ff-4143-b2b7-6a1a069998dc)

![image](https://github.com/microsoft/terminal/assets/39698/abcb4f43-c6f0-40b3-a847-77667aa80016)

**Comment by Summerh15** [comment]
.github/workflows/generator-generic-ossf-slsa3-publish.yml

**Comment by RedTail72** [comment]
I'm not sure if this is what #16196 is addressing, but I decided to run Process Monitor while starting up Terminal and saw it was trying to load `C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.17.11461.0_x64__8wekyb3d8bbwe\CascadiaMono.ttf`, but I'm currently running v1.18.2822.0.  Just thought I'd share incase the PR doesn't address this.

![image](https://github.com/microsoft/terminal/assets/15264370/5aa65165-d450-484d-89f6-f2c7fca56605)
![image](https://github.com/microsoft/terminal/assets/15264370/66dcbcd9-d712-4672-baf8-53db6f3e89e5)

**Comment by RedTail72** [comment]
> I'm not sure if this is what #16196 is addressing, but I decided to run Process Monitor while starting up Terminal and saw it was trying to load `C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.17.11461.0_x64__8wekyb3d8bbwe\CascadiaMono.ttf`, but I'm currently running v1.18.2822.0. Just thought I'd share incase the PR doesn't address this.
> 
> 
I should have waited to post until I did this, but after restarting my machine, I am no longer having issues with the _Cascadia Mono font not found_ message.  My font is still set to Cascadia Mono, just no more error messages.

**Comment by eduarddejong** [comment]
> > I'm not sure if this is what #16196 is addressing, but I decided to run Process Monitor while starting up Terminal and saw it was trying to load `C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.17.11461.0_x64__8wekyb3d8bbwe\CascadiaMono.ttf`, but I'm currently running v1.18.2822.0. Just thought I'd share incase the PR doesn't address this.
> 
> I should have waited to post until I did this, but after restarting my machine, I am no longer having issues with the _Cascadia Mono font not found_ message. My font is still set to Cascadia Mono, just no more error messages.

I can confirm your experience. I haven't booted this Windows system the last days. So my system started updating several "things".
After a while I opened the Terminal and god this strange error message. So I checked Github here.

Before rebooting, I could enter the exact directory that you mention in my Explorer. The font files where not possible to open ad all. Double click failed. But also a thing like trying to open it with Notepad.
It gave me a permissions related error.

After rebooting, this path did not exist anymore, but the Terminal was working fine again and the font just loads without any issues.

So my conclusion is that the Terminal still tried to load the font from an old directory that was being uninstalled by the update. But somehow that process was not fully complete until reboot.

Which may not be strange at all, but it's not what you expect from an App Store based app in general.

So it seems that some kind of proper warning system warning the user that he really needs to reboot is missing here.
